python/lsst/ts/MTMount/mock/controller.py

```diff
@@ -38,6 +41,19 @@
 from .top_end_chiller_device import TopEndChillerDevice
 
 
+async def wait_tasks(*tasks):
+    """Wait for one or more tasks to finish; if one fails cancel the others.
```

Comment:
So this is not exactly what is going on here. What this method does is wait for all tasks to finish. If any exception occurs it will cancel all tasks and raise the exception.